core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -155,51 +211,91 @@ object ConsumerGroupCommand {
         deleteAllForTopic()
     }
 
-    protected def describeGroup(group: String) {
+    protected def collectGroupAssignment(group: String): Option[Array[PartitionAssignmentState]] = {
       val props = if (opts.options.has(opts.commandConfigOpt)) Utils.loadProps(opts.options.valueOf(opts.commandConfigOpt)) else new Properties()
       val channelSocketTimeoutMs = props.getProperty("channelSocketTimeoutMs", "600").toInt
       val channelRetryBackoffMs = props.getProperty("channelRetryBackoffMsOpt", "300").toInt
-      val topics = zkUtils.getTopicsByConsumerGroup(group)
-      if (topics.isEmpty)
-        println("No topic available for consumer group provided")
-      printDescribeHeader()
-      topics.foreach(topic => describeTopic(group, topic, channelSocketTimeoutMs, channelRetryBackoffMs))
-    }
+      if (!zkUtils.getConsumerGroups().contains(group))
+        return None
 
-    private def describeTopic(group: String,
-                              topic: String,
-                              channelSocketTimeoutMs: Int,
-                              channelRetryBackoffMs: Int) {
-      val topicPartitions = getTopicPartitions(topic)
-      val groupDirs = new ZKGroupTopicDirs(group, topic)
-      val ownerByTopicPartition = topicPartitions.flatMap { topicPartition =>
-        zkUtils.readDataMaybeNull(groupDirs.consumerOwnerDir + "/" + topicPartition.partition)._1.map { owner =>
-          topicPartition -> owner
+      val topics = zkUtils.getTopicsByConsumerGroup(group)
+      var assignmentRows: Array[PartitionAssignmentState] = Array()
+      val topicPartitions = getAllTopicPartitions(topics)
+      var groupMemberIds = zkUtils.getConsumersInGroup(group)
+
+      // mapping of topic partition -> ephemeral owner
+      val eOwnerByTopicPartition = mutable.Map[TopicAndPartition, Long]()
+      topicPartitions.foreach { topicPartition =>
+        val data = zkUtils.readDataMaybeNull(new ZKGroupTopicDirs(group, topicPartition.topic).consumerOwnerDir + "/" + topicPartition.partition)
+        val owner = data._1
+        if (!owner.isEmpty) {
```

Comment:
Maybe change this to `map`, then I think `flatMap` as suggested above would work nicely.